tools/data_source/upload.py

```diff
@@ -76,7 +82,7 @@ def add_file(dataset, registry, json_file, output_path):
     line_count = None
     converted_path = None
     stdout = None
-    link_data_only = dataset.get('link_data_only', 'copy_files')
+    link_data_only = dataset.get('link_data_only', 'copy_files') != 'copy_files'
```

Comment:
I think we should check that the value is in ('link_data_only', 'copy_files').

Reply:
I've included this check in https://github.com/galaxyproject/galaxy/pull/5220 - I'm not sure that PR will make 18.01 so if you believe this is urgent I can backport it into its own PR.

Reply:
That's fine for me, thanks @jmchilton!